src/GlobalDiffEq.jl

```diff
@@ -25,6 +27,71 @@ function DiffEqBase.__solve(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase
     return sol
 end
 
-export GlobalRichardson
+struct GlobalAdjoint{A} <: GlobalDiffEqAlgorithm
+    alg::A
+end
+
+function DiffEqBase.__solve(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.AbstractDAEProblem},
+                            alg::GlobalAdjoint, args...;
+                            dt,kwargs...)
+
+    opt = Dict(kwargs)
+    otheropts = delete!(copy(opt), :dt)
+    tstops = get(opt, :tstops, range(prob.tspan[1], stop=prob.tspan[2], step=dt))
+    local sol
+
+    n = length(prob.u0)
+    k = 2 # 2 random vectors
+    C = 11/6 # for k = 2
+    E₂ = 2/π
+    Eₙ = vec_expt_full(n)
+    gtol_get = get(opt, :abstol, 1e-6) # target gtol is abstol? abstol cannot be zero?
+    z₁, z₂ = orth_vec(n)
+
+    function adjoint_loss_z₁(u0, p)
+        # g(u), reverse-mode adjoint
+        _prob = remake(prob, u0=u0, p=p)
+        _sol = solve(_prob, alg.alg; dt=dt, sensealg=DiffEqSensitivity.QuadratureAdjoint())
+        z₁' * _sol.u[end]
+    end
+
+    function adjoint_loss_z₂(u0, p)
+        # g(u), reverse-mode adjoint
+        _prob = remake(prob, u0=u0, p=p)
+        _sol = solve(_prob, alg.alg; dt=dt, sensealg=DiffEqSensitivity.QuadratureAdjoint())
+        z₂' * _sol.u[end]
+    end
```

Comment:
In this form it's only the error with respect to the final time point. That's 😅 not sure how to avoid that though.

Reply:
Could this instead be generalized to an outer product? I'd have to work it out.

Reply:
Revised this loss function.